_posts/2016-05-11-CocoaPods-App-1.0.md

```diff
@@ -0,0 +1,68 @@
+
+So the CocoaPods app hosts everything. Literally, it has it's own self-contained copies of `bzr`, `curl`, `git`, `mercurial`, `ncurses`, `openssl`, `pkg-config`, `ruby`, `scons`, `serf`, `subversion`, `yaml` and `zlib`. This allows the CocoaPods team to ensure consitant behavior across all user setups.
+
+This concept is not new for Cocoa developers, as Xcode has been embedding it's own dependencies inside it's `.app` since it moved to the Mac App Store. The advantage from a user's perspective is that it works, and you can install your own crazy version of git, or curl or ruby, and CocoaPods still works.
+
+## What does it do well?
+
+Glad you asked, 'cause it does two things really well:
+
+* Running `pod install` or `pod update`
+* Editing a Podfile
+
+That was basically the spec for the app from day 1. Once [@alloy](https://twitter.com/alloy/) had proved that `pod install/update` was working, [@nate_west](https://twitter.com/nate_west) and I took over and started working on the latter.
+
+### Podfile Editing
+
+Some of these features are mundane, _"hey, wow, syntax highlighting&lt;/sarcasm>"_ but others are really fascinating deep levels of context-specific integration that you'd expect from an IDE.
+
+We've started out with syntax highlighting, and then added useful examples of auto-completion results that show how to use the CocoaPods DSL.
+
+{% breaking_image /assets/blog_img/app-1.0/useful-autocomplete.png, /assets/blog_img/app-1.0/useful-autocomplete.png %}
+
+The app will autocomplete your Pods, including private pods. This feels magical when you don't expect it. We've been having discussions about ways to introduce more of the internal CocoaPods knowledge into the [auto-complete like this](https://github.com/CocoaPods/CocoaPods-app/issues/311) - you're welcome to help out.
+
+{% breaking_image /assets/blog_img/app-1.0/autocomplete-pods.png, /assets/blog_img/app-1.0/autocomplete-pods.png  %}
+
+The app will generate an overview of what your integration will look like without needing to be installed, so you can understand what CocoaPods _will_ do before it has done it.
+
+{% breaking_image /assets/blog_img/app-1.0/integration-overview.png, /assets/blog_img/app-1.0/integration-overview.png  %}
+
+The app provides a UI for updating your Specs repos, you can find out a little bit more about why this is needed in our [Master spec-repo rate limiting post‑mortem](/Master-Spec-Repo-Rate-Limiting-Post-Mortem).
+
+{% breaking_image /assets/blog_img/app-1.0/update-specs.png, /assets/blog_img/app-1.0/update-specs.png %}
+
+It also has the ability to start a new CocoaPods project, or to even remove CocoaPods from a project entirely. I don't want to give away all of the app's secrets though, so you should give it a test run.
```

Comment:
there's two big CTAs under it - I'm not worried there

![screen shot 2016-05-11 at 11 36 08 am](https://cloud.githubusercontent.com/assets/49038/15186823/90ed9490-176c-11e6-99bb-c6c4593f1a47.png)